### PR TITLE
vulkan-validationlayers: add 1.3.239.0 + modernize more for conan v2

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/16480

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
draft until https://github.com/conan-io/conan-center-index/pull/16523, https://github.com/conan-io/conan-center-index/pull/16261 & https://github.com/conan-io/conan-center-index/pull/16617 are merged

**Comment by SpaceIm** [comment]
merge was unecessary ;) I have to bump cmake anyway to pass v2 pipeline

**Comment by RubenRBS** [comment]
Hah, you're faster than me then, I just wanted to save you some waiting time but didn't realize cmake was out of date :)

**Comment by RubenRBS** [comment]
I still don't know enough about MacOS to know if this hook warning is real or a false positive, happy to get some info on it :)

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 1]
```diff
@@ -96,9 +96,13 @@ def layout(self):
 
     def requirements(self):
         self.requires("robin-hood-hashing/3.11.5")
-        # TODO: set private=True, once the issue is resolved https://github.com/conan-io/conan/issues/9390
-        self.requires(self._require("spirv-tools"), private=not hasattr(self, "settings_build"))
-        self.requires(self._require("vulkan-headers"))
+        self.requires(self._require("spirv-headers"))
+        if Version(conan_version).major < "2":
+            # TODO: set private=True, once the issue is resolved https://github.com/conan-io/conan/issues/9390
```
Comment (prince-chrismc) [inline recipes/vulkan-validationlayers/all/conanfile.py:101]: 🤔 there very little reason things should be exposed like this... its an old PR but could you elaborate? 🙏

File: recipes/vulkan-validationlayers/all/conanfile.py [hunk 2]
```diff
@@ -131,13 +135,25 @@ def loose_lt_semver(v1, v2):
         if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "5":
             raise ConanInvalidConfiguration("gcc < 5 is not supported")
 
+    def _cmake_new_enough(self, required_version):
```
Comment (prince-chrismc) [inline recipes/vulkan-validationlayers/all/conanfile.py:138]: Should not be required with the new portable binaries

**Review by prince-chrismc — COMMENTED** [review]
Just 1 questions :)

File: recipes/vulkan-validationlayers/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/vulkan-validationlayers/all/conanfile.py:101]: I don't remember, and I'm not even sure do have written this specific block.

**Comment by SpaceIm** [comment]
> I still don't know enough about MacOS to know if this hook warning is real or a false positive, happy to get some info on it :)

There is a detailed explanation here: https://github.com/conan-io/hooks/issues/376

For the specific case of vulkan-validationlayers, it doesn't matter because this module file is not intended to be linked but loaded dynamically at runtime by vulkan-loader. Hook could be improved to not check dylib files not listed in cpp_info.libdirs

File: recipes/vulkan-validationlayers/all/patches/1.3.239.0-0001-fix-cmake.patch
Comment (juan-lunarg) [inline recipes/vulkan-validationlayers/all/patches/1.3.239.0-0001-fix-cmake.patch:1]: FWIW I've updated the CMake to no longer need this patch as of https://github.com/KhronosGroup/Vulkan-ValidationLayers/commit/0d83a6e66cd6b71dcea656a06ad60093313d6a27

File: recipes/vulkan-validationlayers/all/conanfile.py (hunk 2 above)
Comment (RubenRBS) [reply recipes/vulkan-validationlayers/all/conanfile.py:138]: Ditto

**Review by RubenRBS — COMMENTED** [review]
Minor request, other than that it looks good. Thanks for your patience :)

File: recipes/vulkan-validationlayers/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/vulkan-validationlayers/all/conanfile.py]: removed in https://github.com/conan-io/conan-center-index/pull/16131/commits/0235e035265f430e38dededc0f3d0293accf07f8

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
I have to fix something, the value of `layer.library_path` in `VkLayer_khronos_validation.json` is incorrect (at least on macOS). It's `libVkLayer_khronos_validation.dylib`, while it should be `../../../lib/libVkLayer_khronos_validation.dylib`.

**Pull request merged**